src/components/TaskTable.vue

```diff
@@ -0,0 +1,55 @@
+<script setup lang="ts">
+import DataTable from 'primevue/datatable';
+import Column from 'primevue/column';
+import Button from 'primevue/button';
+import { type TaskItem } from '../services/TaskService';
+import { RouterLink } from 'vue-router';
+import { defineProps } from 'vue';
```

Comment:
The defineProps import is redundant because defineProps is automatically available in <script setup> without needing to import it. This line can be removed.